### ISSUE TITLE
Text handling - too much, too many places. Defaults?

### ISSUE THREAD

**Issue by NiklasBr** [JohnCoates/Aerial#993]
#### General troubleshooting tips

Before logging an issue please check that:
- [x] You have the latest version installed (There may be a beta version that fixes your issue), see here for the latest releases and bug fixes: https://github.com/JohnCoates/Aerial/releases
- [x] Your issue isn't already mentioned in our [issues](https://github.com/JohnCoates/Aerial/issues). You may find a workaround there or a similar request already made.
- [x] Your problem isn't mentioned in the [troubleshooting page](https://github.com/JohnCoates/Aerial/blob/master/Documentation/Troubleshooting.md).

If none of this fixes your issue, tell us about the problem you are experiencing or the feature you'd like to request.

#### Required information
In order to help us sort your issue, we ask that you provide the following information:
- [x] Mac model: iMac 27 inch
- [x] macOS version: 10.13.6
- [x] Monitor setup: Just the built-in one

If appropriate, please enable `Debug mode` and `Log to disk` in `Advanced` tab and replicate your bug, then attach the `AerialLog.txt` file (You can access this file through the Advanced tab).

#### Description of issue / Feature request

In the *Info* tab there is a *Text settings* alternative, but it only deals with shadow and fade, I would expect fonts to be controlled from that location as well. I find it cumbersome and a bit annoying to have to set the same settings in four different places, why can't I control text settings from the *Text settings* tab?

I also wonder what the *Reset font to default* does, nothing happens when I click it, it just stays on the font selected. Where can I change the default font? A default setting would indeed be nice.

**Comment by glouel** [comment]
Hi @NiklasBr 

> In the _Info_ tab there is a _Text settings_ alternative, but it only deals with shadow and fade, I would expect fonts to be controlled from that location as well. I find it cumbersome and a bit annoying to have to set the same settings in four different places, why can't I control text settings from the _Text settings_ tab?

As to why it's like that, it wasn't in the old days, but people wanted to be able to pick a different font/size for some of their Infos. Some people wanted a fixed width font for clock and proportional for others, etc. The default settings use the same font everywhere (Helvetica Neue) but with different sizes FYI (and sometimes a thinner variant).

I do agree it's a bit cumbersome if you want the same thing everywhere, the thing is, font name and size are tied in the same font picker, so I could have a "default override" but that would include both font and size. I'm not sure that's great, unless you don't care about size? 

> I also wonder what the _Reset font to default_ does, nothing happens when I click it, it just stays on the font selected. Where can I change the default font? A default setting would indeed be nice.

It resets to Aerial's default preset, which is Helvetica Neue in either Medium or Thin variant, and different sizes that I arbitrarily picked as pleasing to me ;)

**Comment by NiklasBr** [comment]
I understand better now. Thanks for this excellent software! This is 100% your decision, and you should not listen to my _bad opinion(tm)_: maybe take an opinionated stance rather than adding everything and the kitchen sink? You can't please all of the people all the time. ;)

Well, I do somwehat care about size, but personally I would not mind if you only had one or two well thought-out sizes available to choose from. So, back to the core of this issue: I don't think I can get the Defaults working if it should reset to Helvetica, it stays on the same, e.g. [IBM Plex](https://www.ibm.com/plex/) (installed with Brew). How can I verify that it works?

**Comment by glouel** [comment]
> I understand better now. Thanks for this excellent software! This is 100% your decision, and you should not listen to my _bad opinion(tm)_: maybe take an opinionated stance rather than adding everything and the kitchen sink? You can't please all of the people all the time. ;)

It's pretty hard to find a good solution every time. I definitely tend to err a lot more on the "sure I'll add one more option" that the opposite. With a few bad experiences in the past, I try to restrain myself but I still try and consider every suggestion to at the very least influence my fairly opinionated decisions ;) Adding a unmute sound button was... not something I expected I would do for example. 

> Well, I do somwehat care about size, but personally I would not mind if you only had one or two well thought-out sizes available to choose from. 

I think this may be the core issue, because I define what's pleasing to me as what's pleasing to me on my 27" iMac, and some fonts may be likely too big on, say, a 13 inch Macbook Air. I do try to double check from time to time on my old non retina air and it looks decent enough, but I can see your point here. 

However just to clarify, what I meant was that if I did add an override of some sort, it would mean every info would share the same font, and same size (that you could still pick). I did think about scaling default font size to screen size when I rewrote the Info system some times back and thought "that's too much" and thought people could set whatever they want anyway to suit their taste, and scaling my taste wasn't needed :)

>So, back to the core of this issue: I don't think I can get the Defaults working if it should reset to Helvetica, it stays on the same, e.g. [IBM Plex](https://www.ibm.com/plex/) (installed with Brew). How can I verify that it works?

First, to be clear, are we talking about the button I circled in red ?
 
![Capture d’écran 2020-05-26 à 14 56 45](https://user-images.githubusercontent.com/37544189/82903427-39e25600-9f61-11ea-8ba1-a87e8256afe3.jpg)

I would suggest trying with the same one (Location). Here I set a random font, and after pressing the red circled button, it goes back to (in this case) : 

![Capture d’écran 2020-05-26 à 14 58 23](https://user-images.githubusercontent.com/37544189/82903618-7dd55b00-9f61-11ea-8a42-54763100b6d2.jpg)

Does that not happen for you ? It's quite possible the button is broken for a given info or something like that, although I think they share the same code so that would be weird. Please let me know what happens when you do this.

**Comment by Amir-Hass** [comment]
Is there an option to hide the location text of the current screensaver shown? I find that it updates too often, and I would like to disable it outright

**Comment by glouel** [comment]
@Amir-Hass of course, see the screenshot just above your post, same place, uncheck the "Enable" for location, this will disable the location text showing.

**Comment by Amir-Hass** [comment]
Hello, thanks for the quick reply I don’t mean the location text for the weather. I mean like say we’re flying over New York City in the screensaver, and it shows the related caption, how can I turn that off? Thanks!

**Comment by glouel** [comment]
@Amir-Hass yeah that's exactly that one. Go to "Info" tab, then pick "Location" on the left, and disable it. 

![82903618-7dd55b00-9f61-11ea-8a42-54763100b6d2](https://user-images.githubusercontent.com/37544189/89670157-01f07b00-d8e1-11ea-8e4b-741c053fc46a.jpg)

See what I circled in purple. It will go red instead of green in front of Locations, and then those won't show at all. You can enable/disable various informations (weather, location [the one you want], clock, etc) this way.

**Comment by Amir-Hass** [comment]
Ohhhh I must be blind, thanks for showing me that!

**Comment by NiklasBr** [comment]
Sorry I'm late to return here

> First, to be clear, are we talking about the button I circled in red ?

Yes

> I would suggest trying with the same one (Location). Here I set a random font, and after pressing the red circled button, it goes back to (in this case) :
> Does that not happen for you ? It's quite possible the button is broken for a given info or something like that, although I think they share the same code so that would be weird. Please let me know what happens when you do this.

I'm unable to make it work. Perhaps because I am stuck on 10.13.6?

**Comment by glouel** [comment]
> > I would suggest trying with the same one (Location). Here I set a random font, and after pressing the red circled button, it goes back to (in this case) :
> > Does that not happen for you ? It's quite possible the button is broken for a given info or something like that, although I think they share the same code so that would be weird. Please let me know what happens when you do this.
> 
> I'm unable to make it work. Perhaps because I am stuck on 10.13.6?

Does the button next to it (to its left) opens the font selection panel ? If you pick something random, close it, does it show to the left of this ? And then pressing reset does nothing ? I'll try to confirm that, I can't think of a reason why it would only be broken on 10.13 or below, but I'll check thanks !

**Comment by NiklasBr** [comment]
Yes, I can change fonts using the cogwheel button, but pressing the reset button will not adjust it to the defaults.

**Comment by glouel** [comment]
> Yes, I can change fonts using the cogwheel button, but pressing the reset button will not adjust it to the defaults.

I tried to reproduce in a VM with 2.0.0 beta, it works for me here, hmm (and I didn't change anything around here I think). You are running 1.9.2 right ? Not something super old 

At least I saw that the icons were missing in the beta for 10.13, I'll fix that : 

![Capture d’écran 2020-08-19 à 17 00 28](https://user-images.githubusercontent.com/37544189/90651848-cf7a4280-e23d-11ea-8a6b-8e29c9698486.jpg)
 
That's the button we're talking about in 2.0.0

**Comment by glouel** [comment]
This has been centralised in v4 thankfully 😅
